pkg/events/client.go

```diff
@@ -0,0 +1,134 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/workos/workos-go/v2/pkg/workos_errors"
+
+	"github.com/workos/workos-go/v2/internal/workos"
+	"github.com/workos/workos-go/v2/pkg/common"
+)
+
+// ResponseLimit is the default number of records to limit a response to.
+const ResponseLimit = 10
+
+// Client represents a client that performs Event requests to the WorkOS API.
+type Client struct {
+	// The WorkOS API Key. It can be found in https://dashboard.workos.com/api-keys.
+	APIKey string
+
+	// The http.Client that is used to get Event records from WorkOS.
+	// Defaults to http.Client.
+	HTTPClient *http.Client
+
+	// The endpoint to WorkOS API. Defaults to https://api.workos.com.
+	Endpoint string
+
+	once sync.Once
+}
+
+func (c *Client) init() {
+	if c.HTTPClient == nil {
+		c.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+	}
+
+	if c.Endpoint == "" {
+		c.Endpoint = "https://api.workos.com"
+	}
+}
+
+// Event contains data about a particular Event.
+type Event struct {
+	// The Event's unique identifier.
+	ID string `json:"id"`
+
+	// The type of Event.
+	Event string `json:"event"`
+
+	// The Event's data in raw encoded JSON.
+	Data json.RawMessage `json:"data"`
+
+	// The Event's created at date.
+	CreatedAt string `json:"created_at"`
```

Comment:
Should this be a string? We seem to use different types in different parts of the SDK. Audit logs [uses `time.Time`](https://github.com/workos/workos-go/blob/ea6316d005c5c9fb4bf0fe03277edf0d9ef32746/pkg/auditlogs/client.go#L74) but organizations [uses a `string`](https://github.com/workos/workos-go/blob/ea6316d005c5c9fb4bf0fe03277edf0d9ef32746/pkg/organizations/client.go#L88).

Reply:
Hmm, yeah we could use `time.Time`, thought we were using `string` across the board but Time fits better here.